dotCMS/src/main/java/com/dotcms/security/secret/SecretsStoreKeyStoreImpl.java

```diff
@@ -0,0 +1,308 @@
+        secretStoreFile.delete();
+        flushCache();
+        return true;
+
+    }
+
+
+    /**
+     * This method saves a secret value into the keystore. The value to be stored is encrypted using the
+     * company key and is used as a "password" for a keystore entry - using it as a passowrd allows us
+     * to store an arbitrary string in a keystore
+     * 
+     * @param variableKey
+     * @param variableValue
+     * @throws NoSuchAlgorithmException
+     * @throws InvalidKeySpecException
+     * @throws KeyStoreException
+     * @throws Exception
+     */
+    @Override
+    public boolean saveValue(final String variableKey, final char[] variableValue) {
+        try {
+            final SecretKeyFactory factory = SecretKeyFactory.getInstance(
+                    SECRETS_STORE_SECRET_KEY_FACTORY_TYPE);
+            final SecretKey generatedSecret = factory.generateSecret(new PBEKeySpec(encrypt(variableValue).toCharArray()));
+            final KeyStore keyStore = getSecretsStore();
+            final PasswordProtection keyStorePP = new PasswordProtection(loadStorePassword());
+            keyStore.setEntry(variableKey, new KeyStore.SecretKeyEntry(generatedSecret), keyStorePP);
+            saveSecretsStore(keyStore);
+            flushCache(variableKey);
+        } catch (Exception e) {
+            Logger.warn(this.getClass(), "Unable to save secret from " + SECRETS_STORE_FILE + ": " + e);
+            throw new DotRuntimeException(e);
+        }
+        return true;
+    }
+
+    /**
+     * deletes a value from the store
+     */
+    @Override
+    public void deleteValue(final String secretKey) {
+        try {
+            final KeyStore keyStore = getSecretsStore();
+            keyStore.deleteEntry(secretKey);
+            saveSecretsStore(keyStore);
+            flushCache(secretKey);
+        } catch (KeyStoreException e) {
+            Logger.warn(this.getClass(), "Unable to delete secret from  " + SECRETS_STORE_FILE + ": " + e);
+            throw new DotRuntimeException(e);
+        }
+    }
+
+    private Key key() {
+        return Sneaky.sneak(() -> APILocator.getCompanyAPI().getDefaultCompany()).getKeyObj();
+    }
+
+    @VisibleForTesting
+    protected String encrypt(final char[] val) {
+        return encrypt(new String(val));
+    }
+
+    @VisibleForTesting
+    protected String encrypt(final String val) {
+        return Sneaky.sneak(() -> Encryptor.encrypt(key(), val));
+    }
+
+    @VisibleForTesting
+    protected String digest(final String val) {
+        return Sneaky.sneak(() -> Encryptor.digest(val));
+    }
+
+    @VisibleForTesting
+    protected char[] decrypt(final String encryptedString) {
+        if (encryptedString == null || encryptedString.length() == 0) {
+            return null;
+        }
+        return Sneaky.sneak(() -> Encryptor.decrypt(key(), encryptedString).toCharArray());
+    }
+
+    @VisibleForTesting
+    protected char[] decrypt(final char[] encryptedString) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)